### DISCUSSION TITLE
Utilize scene.turn_on instead of turning on ALL lights when presence detected

### DISCUSSION THREAD

**Discussion by rsnodgrass** [jseidl/magic-areas#217] [Ideas / Feature Requests]
Currently magic_areas assumes that ALL lights assigned to a room should be turned on to full upon entry to the room. This will leverage HA's ever improving scene support. If a scene is defined, this scene would override the default of light.turn_on/off behavior of magic_areas and use the scene instead.

Ideally, a scene could be assigned for room entry that allows setting which lights to turn on, at what dimmable level, as well as other devices in a room that might be turned on (e.g. a fan, or climate setting). For leaving a room, a second scene would be needed to set the "away" settings for a room.

I think three scenes would be needed:

* Presence - scene activated when presence detected (when not sleeping)
* Goodbye - scene activated when presence no longer detected
* Sleep - scene activated when presence detected during sleep_mode

These would need to be optionally assigned to auto_areas rooms instead of the default lights.turn_on/turn_off, since I don't think programmatically defined scenes can be edited within UI to assign lights/etc.

I believe Scenes can be assigned to Areas in HA, so with consistent naming (e.g. includes one of the above terms like Sleep), so that might be a way to "automate" this.

**Comment by jseidl** [comment]
Hi, are you on the current release or in the beta one?

The beta release has support for different light groups which might solve one of your concerns (turning on all lights).

As per the brightness, the main idea is to use [Adaptive Lighting](https://github.com/basnijholt/adaptive-lighting) to control that.

Give the beta release a try with Adaptive Lighting and let me know if that addresses the needs. Scene support can be a feature request but isn't on the backlog as of now.

**Comment by jseidl** [comment]
BTW as of 3.0 Magic Areas no longer turns on ALL lights if you don't specify one. Only light groups listening to the `occupied` will turn on.

**Comment by zazu-sf** [comment]
+1
It would be great if you could start scenes instead of all lights or groups of lights. This would allow you to do a bit more directly in Magic Area and you wouldn't need another component. You could select lights, brightness and color directly in Magic Areas by starting a scene.

**Comment by jseidl** [comment]
Hi folks, I just noted this is a duplicate from https://github.com/jseidl/hass-magic_areas/discussions/188. Can you folks upvote the older one so I can keep track of it? Closing as duplicate, tracking on the older one.
